src/app/qgstemplateprojectsmodel.cpp

```diff
@@ -0,0 +1,103 @@
+/***************************************************************************
+
+               ----------------------------------------------------
+              date                 : 16.5.2019
+              copyright            : (C) 2019 by Matthias Kuhn
+              email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgstemplateprojectsmodel.h"
+#include "qgsziputils.h"
+#include "qgssettings.h"
+#include "qgsapplication.h"
+#include "qgis.h"
+#include "qgsrecentprojectsitemsmodel.h"
+
+#include <QStandardPaths>
+#include <QDir>
+#include <QCryptographicHash>
+#include <QPainter>
+
+#include <memory>
+
+
+QgsTemplateProjectsModel::QgsTemplateProjectsModel( QObject *parent )
+  : QStandardItemModel( parent )
+{
+  const QStringList paths = QStandardPaths::standardLocations( QStandardPaths::AppDataLocation );
+  QString templateDirName = QgsSettings().value( QStringLiteral( "qgis/projectTemplateDir" ),
+                            QgsApplication::qgisSettingsDirPath() + QStringLiteral( "project_templates" ) ).toString();
+
+  for ( const QString &templatePath : paths )
+  {
+    const QString path = templatePath + QDir::separator() + QStringLiteral( "project_templates" );
+    scanDirectory( path );
+    mFileSystemWatcher.addPath( path );
+  }
+
+  scanDirectory( templateDirName );
+  mFileSystemWatcher.addPath( templateDirName );
+
+  connect( &mFileSystemWatcher, &QFileSystemWatcher::directoryChanged, this, &QgsTemplateProjectsModel::scanDirectory );
+
+  setColumnCount( 1 );
+}
+
+void QgsTemplateProjectsModel::scanDirectory( const QString &path )
+{
+  QDir dir = QDir( path );
+  const QFileInfoList files = dir.entryInfoList( QStringList() << QStringLiteral( "*.qgs" ) << QStringLiteral( "*.qgz" ) );
```

Comment:
Is this case insensitive by default?

Reply:
I guess. NoFiter is specified by default and `QDir::CaseSensitive` is listed in filters.